src/llvm.cr

```diff
@@ -91,12 +91,6 @@ module LLVM
   def self.default_target_triple : String
     chars = LibLLVM.get_default_target_triple
     case triple = string_and_dispose(chars)
-    when .starts_with?("x86_64-apple-macosx"), .starts_with?("x86_64-apple-darwin")
```

Comment:
Isn't this still needed to remove the deployment version?

Reply:
I deliberately remove this because we want to **keep** the deployment version. Since not specifying it, causes linker warnings as of Xcode 15. See https://github.com/crystal-lang/crystal/issues/13846#issuecomment-2040146898 for details.